src/net/discovery/config.rs

```diff
@@ -0,0 +1,204 @@
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct UrlVisitor;
+        impl<'de> serde::de::Visitor<'de> for UrlVisitor {
+            type Value = Origin;
+
+            fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+                formatter.write_str("a valid format origin made up of http[s]://<authority>")
+            }
+
+            fn visit_str<E: serde::de::Error>(self, v: &str) -> Result<Self::Value, E> {
+                let uri = v.parse::<Uri>().map_err(|err| E::custom(err.to_string()))?;
+                let parts = uri.into_parts();
+                let scheme = parts.scheme.ok_or_else(|| E::custom("missing scheme"))?;
+                let authority = parts
+                    .authority
+                    .ok_or_else(|| E::custom("missing authority"))?;
+                Ok(Origin { scheme, authority })
+            }
+        }
+        deserializer.deserialize_str(UrlVisitor)
+    }
+}
+
+/// Configuration values relevant for interacting with other helpers
+#[derive(Serialize, Deserialize)]
+struct Peer {
+    origin: Origin,
+    public_key: PublicKey,
+}
+
+/// All config value necessary to discover other peer helpers of the MPC ring
+#[derive(Serialize, Deserialize)]
+struct Config {
+    peers: [Peer; 3],
+}
+
+impl Config {
+    /// Reads config from `file_location`. Expects file to be json format
+    /// # Errors
+    /// if the file does not exist, or is in an invalid format
+    #[allow(dead_code)] // TODO: will use in upcoming PR
+    pub fn new(file_location: PathBuf) -> Result<Self, Error> {
+        use std::{fs::File, io::BufReader};
+
+        let contents = File::open(file_location)?;
+        let buffered = BufReader::new(contents);
+        Ok(serde_json::from_reader(buffered)?)
+    }
+}
+
+impl PeerDiscovery for Config {
+    /// TODO: support HTTPS
+    fn peers(&self) -> [MpcHelperClient; 3] {
+        self.peers
+            .iter()
+            .map(|peer| MpcHelperClient::new(peer.origin.clone().into()))
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap_or_else(|_| panic!("could not convert vec into array"))
+    }
+
+    // TODO: do we need different public key for HTTP and PRSS, or can they share?
+    fn prss(&self, role: Identity) -> prss::Endpoint {
+        let mut rng = thread_rng();
+        let endpoint = prss::Endpoint::prepare(&mut rng);
+        endpoint.setup(
+            &self.peers[role.peer(Direction::Left)].public_key,
+            &self.peers[role.peer(Direction::Right)].public_key,
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rand::distributions::Alphanumeric;
+    use rand::{thread_rng, Rng};
+    use std::fs::File;
+    use std::io::Write;
+
+    #[allow(unused)] // TODO: how to test prss seed equality?
+    const ROLE: Identity = Identity::H2;
+    #[allow(unused)] // TODO: how to test prss seed equality?
```

Comment:
just generate values at the same index - two different prss instances with the same seed will yield the same value